corehq/apps/users/models.py

```diff
@@ -2489,6 +2490,16 @@ def get_sql_locations(self, domain=None):
     def get_location(self, domain):
         return self.get_sql_location(domain)
 
+    def get_usercase_by_domain(self, domain):
+        return CommCareCase.objects.get_case_by_external_id(domain, self._id, USERCASE_TYPE)
```

Comment:
it would be good to add a cached accessor here too:

(copying from the commcare user version)
```python
    @quickcache(['self._id', 'domain'], lambda _: settings.UNIT_TESTING)
    def get_usercase_id(self, domain):
        case = self.get_usercase_by_domain(domain)
        return case.case_id if case else None
```

Reply:
okay this has been changed!